### PR TITLE
chore(typescript-estree): clarify default file extensions

### PR DESCRIPTION
**Pull request by dimitropoulos**
I was doing a few things wrong that encouraged me, whilst debugging an issue with the parser, to make these changes.

I was using `"extraFileExtensions": ["tsx"]` in my code.   For one thing, `tsx` is already scanned.  For another thing, the format of this (which mirrors [`path.extname`](https://nodejs.org/api/path.html#path_path_extname_path)) is such that the `.` is required to work properly.

Once I read the source code and better understood how it really works, I decided that a simple mention of the defaults, as well as an explicit example, could have prevented me from going astray.

### CODE REVIEW

**Comment by dimitropoulos** [comment]
@JamesHenry ok.  I added _something_, but I have to say I'm not up to speed on this project so I:

- haven't/can't run the project to ensure my code works (I just did it "by eye")
- didn't include tests (I would like to, but I don't think I'll have time)
- don't know what the performance implications are (note a nested loop in the check for a file included that is already a default).
- don't know if the top of `getASTFromProject` is the best place to add this kind of code, which, seems like it should be a single responsibility of a _different_ function.

Hopefully that's at least a start?  Sorry I can't do more right now (I need to focus on a separate problem I'm experiencing with the parser that's holding up my team).

**Comment by bradzacher** [comment]
you will have to modify the unit tests in `typescript-estree/tests/parse.ts` appropriately.

you can add tests for your new code in there as well.

**Comment by dimitropoulos** [comment]
@bradzacher I'm sorry, I won't be able to do more with this out of lack of time.  If you would like me to remove the code and just update the documentation please let me know and I will happy do so.

File: packages/typescript-estree/src/parser.ts [hunk 1]
```diff
@@ -99,6 +99,19 @@ function getASTFromProject(
     ];
     let hasMatchedAnError = false;
 
+    extra.extraFileExtensions.forEach(extraExtension => {
+      if (extraExtension.startsWith('.')) {
```
Comment (armano2) [inline packages/typescript-estree/src/parser.ts]: this should be
```suggestion
      if (!extraExtension.startsWith('.')) {
```

**Comment by armano2** [comment]
we should add test cases for this

**Pull request merged**